src/common/webpack/config.ts

```diff
@@ -87,18 +87,8 @@ export function webpackConfigFactory(
                   level: 'verbose',
               }
             : undefined,
-        experiments: isEnvDevelopment
-            ? {
-                  lazyCompilation: config.lazyCompilation
-                      ? {
-                            backend: {
-                                client: require.resolve('./lazy-client.js'),
-                                listen: config.lazyCompilation,
-                            },
-                        }
-                      : undefined,
-              }
-            : undefined,
+
+        experiments: isEnvDevelopment ? configureExperiments(helperOptions) : undefined,
```

Comment:
if extract all experiment settings to a function then environment check should be in that function too. Some experimental settings can be used in prod as well.